### PR TITLE
Covenant Vendor PR

### PR DESCRIPTION
**Pull request by DexterDude**
It's a bit of stuff that generally revolves around one thing, COVENANT VENDORS. Yaay! These vendors are species locked after me using 1% of my tired brain to understand XO's code gibberish. This PR also contains actual coded food instead of just having the name changed of a new [whatever]. Some minor map changes to accommodate this too. Most notably these are: Jiralhanae Quarters now leads to the right instead of into Sangheili quarters to further diversify them, drone carbons now have their own spawn room where Unggoy methane tank storage used to be due to it no longer being in use. Really that's it.

:cl: Dexterdude
rscadd: Adds Covenant Vendors
rscadd: Adds Covenant Food
rscadd: Adds minor changes to the Covenant ship to fit with Covenant Vendors
rscadd: Gave carbon drones their own spawn near Methane Suite as that room isn't used anymore due to changes
/:cl:

### CODE REVIEW

**Comment by DexterDude** [comment]
i think I did it this time

File: code/modules/halo/structures/covvendors.dm [hunk 1]
```diff
@@ -0,0 +1,236 @@
+/obj/machinery/vending/armory/covenant
+	icon = 'code/modules/halo/icons/machinery/covenant/covendor.dmi'
+	var/covenant_secure = 0
+	ai_access_level = 3
+
+/obj/machinery/vending/armory/attackby(var/atom/A,var/mob/user)
```
Comment (X0-11) [inline code/modules/halo/structures/covvendors.dm]: This proc isn't needed. All of this is done by the parent /armory

File: code/modules/halo/structures/covvendors.dm [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+	if(h.species.type == /datum/species/unggoy)
+		return 1
+	return 0
+
+/obj/machinery/vending/armory/covenant/sangheili/weapon // Both ammo, and guns!
+	name = "Covenant - Sangheili Weapon Vendor"
+	desc = "Storage for Covenant Sangheili weapons and ammo"
+	icon_state ="covendor" // SPRITES
+	icon_deny = "covendor-deny"
+	products = list(/obj/item/weapon/gun/energy/plasmapistol = 25,/obj/item/weapon/gun/energy/plasmarifle = 5,/obj/item/weapon/gun/projectile/type31needlerifle = 3
```
Comment (X0-11) [inline code/modules/halo/structures/covvendors.dm]: Amount of plasma pistols seems too high, amount of needler ammo seems low.

File: code/modules/halo/structures/covvendors.dm [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+		,/obj/item/weapon/gun/projectile/type51carbine = 3,/obj/item/ammo_magazine/rifleneedlepack = 12,/obj/item/ammo_magazine/type51mag = 24,/obj/item/weapon/grenade/plasma = 12)
+
+/obj/machinery/vending/armory/covenant/sangheili/equipment // Equipment for Sangheili
+	name = "Covenant - Sangheili Equipment Vendor"
+	desc = "Storage for Covenant Sangheili equipment"
+	icon_state ="covendor" // SPRITES
+	icon_deny = "covendor-deny"
+	products = list(/obj/item/weapon/storage/backpack/sangheili = 10,/obj/item/weapon/storage/belt/covenant_medic = 5,/obj/item/weapon/storage/belt/covenant_ammo = 5
+		,/obj/item/clothing/accessory/storage/bandolier/covenant = 6,/obj/item/clothing/head/helmet/sangheili/ranger = 9,/obj/item/clothing/suit/armor/special/combatharness/ranger = 9
+		,/obj/item/clothing/shoes/sangheili/ranger = 7, /obj/item/clothing/gloves/thick/sangheili/ranger = 7,/obj/item/weapon/tank/air/covenant = 9,/obj/item/clothing/mask/breath = 7
```
Comment (X0-11) [inline code/modules/halo/structures/covvendors.dm]: What's the basis for these ranger armor numbers?

File: code/modules/halo/structures/covvendors.dm [hunk 4]
```diff
@@ -0,0 +1,236 @@
...
+	icon_state ="covendor" // SPRITES
+	icon_deny = "covendor-deny"
+	products = list(/obj/item/weapon/gun/launcher/grenade/brute_shot = 2,/obj/item/weapon/grenade/brute_shot = 8)
+
+/obj/machinery/vending/armory/covenant/kigyar/weapon // Both ammo, and guns!
+	name = "Covenant - Kig-Yar Weapon Vendor"
+	desc = "Storage for Covenant Kig-Yar weapons and ammo"
+	icon_state ="covendor" // SPRITES
+	icon_deny = "covendor-deny"
+	products = list(/obj/item/weapon/gun/energy/plasmapistol = 25,/obj/item/weapon/gun/projectile/needler = 1,/obj/item/weapon/gun/projectile/type51carbine = 6
```
Comment (X0-11) [inline code/modules/halo/structures/covvendors.dm]: Once again, plaspistols too high

File: code/modules/halo/structures/covvendors.dm [hunk 5]
```diff
@@ -0,0 +1,236 @@
...
+	icon_state ="covendor" // SPRITES
+	icon_deny = "covendor-deny"
+	products = list(/obj/item/weapon/gun/energy/beam_rifle = 3)
+
+/obj/machinery/vending/armory/covenant/kigyar/ranger // Equipment for Kig-Yar
+	name = "Covenant - Kig-Yar Ranger Vendor"
+	desc = "Storage for Covenant Kig-Yar ranger equipment"
+	icon_state ="covendor" // SPRITES
+	icon_deny = "covendor-deny"
+	products = list(/obj/item/clothing/head/helmet/ranger_kigyar = 12,/obj/item/clothing/under/covenant/ranger_kigyar = 12,/obj/item/clothing/suit/armor/covenant/ranger_kigyar = 12
```
Comment (X0-11) [inline code/modules/halo/structures/covvendors.dm]: Please provide reasoning for this many ranger suits

File: code/modules/halo/structures/covvendors.dm [hunk 6]
```diff
@@ -0,0 +1,236 @@
...
+		,/obj/item/clothing/head/helmet/ranger_tvoan = 12,/obj/item/clothing/under/covenant/ranger_tvoan = 12,/obj/item/clothing/suit/armor/covenant/ranger_tvoan = 12
+		,/obj/item/weapon/tank/air/covenant = 12,/obj/item/clothing/mask/breath = 12)
+
+/obj/machinery/vending/armory/covenant/unggoy/weapon // Both ammo, and guns!
+	name = "Covenant - Unggoy Weapon Vendor"
+	desc = "Storage for Covenant Unggoy weapons and ammo"
+	icon_state ="covendor" // SPRITES
+	icon_deny = "covendor-deny"
+	products = list(/obj/item/weapon/gun/energy/plasmapistol = 25,/obj/item/weapon/gun/projectile/needler = 3,/obj/item/ammo_magazine/needles = 24
```
Comment (X0-11) [inline code/modules/halo/structures/covvendors.dm]: needler count seems low, plaspistols way too high

File: code/modules/halo/structures/covvendors.dm [hunk 7]
```diff
@@ -0,0 +1,236 @@
...
+/obj/machinery/vending/armory/covenant/unggoy/equipment // Equipment for Unggoy
+	name = "Covenant - Unggoy Equipment Vendor"
+	desc = "Storage for Covenant Unggoy equipment"
+	icon_state ="covendor" // SPRITES
+	icon_deny = "covendor-deny"
+	products = list(/obj/item/weapon/storage/belt/covenant_medic = 5,/obj/item/weapon/storage/belt/covenant_ammo = 5,/obj/item/clothing/accessory/storage/bandolier/covenant = 6
+		,/obj/item/weapon/pickaxe/plasma_drill = 1,/obj/item/weapon/plastique/covenant = 4,/obj/item/weapon/armor_patch/cov = 12)
+
+/obj/machinery/vending/armory/covenant/unggoy/heavy // Equipment for Unggoy
+	name = "Covenant - Kig-Yar Equipment Vendor"
```
Comment (X0-11) [inline code/modules/halo/structures/covvendors.dm]: Incorrect name

File: code/modules/halo/structures/covvendors.dm [hunk 8]
```diff
@@ -0,0 +1,236 @@
...
+	icon_state ="covendor" // SPRITES
+	icon_deny = "covendor-deny"
+	products = list(/obj/item/weapon/gun/projectile/fuel_rod_launcher = 1,/obj/item/ammo_magazine/fuel_rod = 2,/obj/item/turret_deploy_kit/plasturret = 2)
+
+/obj/machinery/vending/armory/covenant/unggoy/supplies
+	name = "Covenant - Unggoy Supplies Vendor"
+	desc = "Storage for Unggoy supplies"
+	icon_state ="covendor" // SPRITES
+	icon_deny = "covendor-deny"
+	products = list(/obj/item/weapon/tank/methane/unggoy_internal = 60)
```
Comment (X0-11) [inline code/modules/halo/structures/covvendors.dm]: This vendor isn't needed. You can't replace internal methane tanks nor do rounds actually go on long enough for them to require refilling.

File: code/modules/halo/structures/covvendors.dm [hunk 9]
```diff
@@ -0,0 +1,236 @@
...
+
+/obj/machinery/vending/armory/covenant/sangheili/food
+	name = "Covenant - Sangheili Food Vendor"
+	desc = "A vendor for Sangheili oriented food."
+	icon_state ="covendor" // SPRITES
+	icon_deny = "covendor-deny"
+	products = list(/obj/item/weapon/reagent_containers/food/snacks/covenant/colo = 20,/obj/item/weapon/reagent_containers/food/snacks/covenant/colo/stew = 15,/obj/item/weapon/reagent_containers/food/snacks/covenant/irukanbar = 60)
```
Comment (X0-11) [inline code/modules/halo/structures/covvendors.dm]: Isn't 60 a bit high?

File: code/modules/halo/structures/covvendors.dm [hunk 10]
```diff
@@ -0,0 +1,236 @@
...
+	icon_state ="covendor" // SPRITES
+	icon_deny = "covendor-deny"
+	products = list(/obj/item/weapon/reagent_containers/food/snacks/covenant/uoi = 20,/obj/item/weapon/reagent_containers/food/snacks/covenant/uoi/stew = 15,/obj/item/weapon/reagent_containers/food/snacks/covenant/irukanbar = 60)
+
+/obj/machinery/vending/armory/covenant/general/food
+	name = "Covenant - Lesser Food Vendor"
+	desc = "A food vendor for the lesser species."
+	icon_state ="covendor" // SPRITES
+	icon_deny = "covendor-deny"
+	products = list(/obj/item/weapon/reagent_containers/food/snacks/covenant/irukanbar = 120)
```
Comment (X0-11) [inline code/modules/halo/structures/covvendors.dm]: 120 is way too many

File: code/modules/halo/structures/covvendors.dm [hunk 11]
```diff
@@ -0,0 +1,236 @@
...
+
+/obj/machinery/vending/armory/covenant/general/medical
+	name = "Covenant - Medical Vendor"
+	desc = "A vendor that supplies medical equipment"
+	icon_state ="covendor" // SPRITES
+	icon_deny = "covendor-deny"
+	products = list(/obj/item/weapon/storage/firstaid/unsc/cov = 5, /obj/item/weapon/storage/firstaid/regular/covenant = 10, /obj/item/weapon/storage/firstaid/fire/covenant = 5
+		,/obj/item/weapon/storage/firstaid/o2/covenant = 5,/obj/item/weapon/storage/firstaid/toxin/covenant = 5,/obj/item/weapon/storage/firstaid/erk/cov = 5,/obj/item/weapon/storage/firstaid/combat/unsc/cov = 5,/obj/item/weapon/storage/firstaid/adv = 5)
```
Comment (X0-11) [inline code/modules/halo/structures/covvendors.dm]: You may want to cross reference the amount of kits with those present on the unsc ship, tweaking up/down where needed.

**Review by X0-11 — DISMISSED** [review]

File: code/modules/halo/structures/covvendors.dm (hunk 3 above)
Comment (DexterDude) [reply code/modules/halo/structures/covvendors.dm]: Current Sangheili role amount is the basis of said ranger armor numbers.

File: code/modules/halo/structures/covvendors.dm (hunk 5 above)
Comment (DexterDude) [reply code/modules/halo/structures/covvendors.dm]: That's how many there were on the ship before I put them in a vendor, roughly. I could reduce it to eight if that would be more viable.

File: code/modules/halo/structures/covvendors.dm (hunk 3 above)
Comment (X0-11) [reply code/modules/halo/structures/covvendors.dm]: Not all sangheili should have a corresponding set of ranger armor. Lower to 1/4 of original value for the ranger armors. They're specialised and not meant for literally everyone.

File: code/modules/halo/structures/covvendors.dm (hunk 5 above)
Comment (X0-11) [reply code/modules/halo/structures/covvendors.dm]: There were twelve of them? I swear I heard more complaining about there being a lack of them. Please confirm this number.

Comment (DexterDude) [reply code/modules/halo/structures/covvendors.dm]: Will do.

File: code/modules/halo/structures/covvendors.dm (hunk 8 above)
Comment (DexterDude) [reply code/modules/halo/structures/covvendors.dm]: Oh. Well then the methane resupply room was kinda pointless in the first place then.

Comment (X0-11) [reply code/modules/halo/structures/covvendors.dm]: Please don't tell me this room had loose methane gas

**Review by X0-11 — APPROVED** [review]

**Review by X0-11 — APPROVED** [review]

**Comment by bloxgate** [comment]
I may be blind, but the PR changelog mentions map changes however there's no changes to the map in the changed files list?

**Pull request merged**